lib/rules/jsx-no-literals.js

```diff
@@ -69,8 +69,8 @@ module.exports = {
     const config = Object.assign({}, defaults, context.options[0] || {});
     config.allowedStrings = new Set(config.allowedStrings.map(trimIfString));
 
-    function defaultMessageId() {
-      if (config.noAttributeStrings) {
+    function defaultMessageId(ancestorIsJSXElement = false) {
```

Comment:
default arguments don't work in node 4
```suggestion
    function defaultMessageId() {
      const ancestorIsJSXElement = arguments.length > 1 && arguments[0];
```